### PR TITLE
refactor: Move Upload from less to css in js

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

- [x] 迁移 Upload 到 css in js
- [x] 原来 Upload 的 dom 结构和样式覆盖非常混乱，现在整体重构了一遍，结构更加精简，样式也更简单。所以 snapshot 也也有大量变化。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Refactor Upload with CSS-in-JS.          |
| 🇨🇳 Chinese |  迁移 Upload 为 CSS in JS 方案。         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/card/style/index.tsx [hunk 1]
```diff
@@ -349,7 +349,7 @@ const genCardStyle: GenerateStyle<CardToken> = (token, hashId): CSSObject => {
 
     [`${componentCls}-hoverable`]: {
       cursor: 'pointer',
-      transition: `box-shadow ${token.motionDurationSlow}s, border-color ${token.motionDurationSlow}s`,
+      transition: `box-shadow ${token.motionDurationSlow}, border-color ${token.motionDurationSlow}`,
```
Comment (afc163) [inline components/card/style/index.tsx:352]: 不应该带 s

File: components/_util/theme/themes/default.ts [hunk 2]
```diff
@@ -73,9 +73,9 @@ export function derivative(token: SeedToken): DerivativeToken {
     ...colorPalettes,
 
     // motion
-    motionDurationFast: `${motionBase + motionUnit * 1}s`,
-    motionDurationMid: `${motionBase + motionUnit * 2}s`,
-    motionDurationSlow: `${motionBase + motionUnit * 3}s`,
+    motionDurationFast: `${(motionBase + motionUnit * 1).toFixed(1)}s`,
+    motionDurationMid: `${(motionBase + motionUnit * 2).toFixed(1)}s`,
+    motionDurationSlow: `${(motionBase + motionUnit * 3).toFixed(1)}s`,
```
Comment (afc163) [inline components/_util/theme/themes/default.ts:78]: 修复 `0.30000000000000004s`

File: components/form/__tests__/__snapshots__/demo.test.js.snap [hunk 3]
```diff
@@ -7501,7 +7501,9 @@ exports[`renders ./components/form/demo/validate-other.md correctly 1`] = `
         <div
           class="ant-form-item-control-input-content"
         >
-          <span>
+          <span
+            class="ant-upload-wrapper"
```
Comment (afc163) [inline components/form/__tests__/__snapshots__/demo.test.js.snap:7505]: 统一增加了 wrapper，方便绑定 hashId 。

**Comment by zombieJ** [comment]
snapshot 也变了，结构变化挪到 feature 上？

**Comment by afc163** [comment]
> snapshot 也变了，结构变化挪到 feature 上？

那就要先重构 4.x 的 Upload 了，成本有点高。

File: components/upload/Upload.tsx [hunk 4]
```diff
@@ -338,6 +341,7 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (pr
           return (
             <UploadList
               prefixCls={prefixCls}
+              className={hashId}
```
Comment (zombieJ) [inline components/upload/Upload.tsx:344]: 最外层有 hashId 为何 UploadList 里还要再加一个 hashId？

File: components/upload/style/dragger.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          display: 'table-cell',
+          verticalAlign: 'middle',
+        },
+
+        [`${componentCls}:not(${componentCls}-disabled):hover`]: {
+          borderColor: token.colorPrimaryHover,
+        },
+
+        [`p${componentCls}-drag-icon`]: {
+          marginBottom: 20,
```
Comment (zombieJ) [inline components/upload/style/dragger.tsx:43]: `FIXME`

File: components/upload/style/dragger.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+          [iconCls]: {
+            color: token.colorPrimary,
+            fontSize: 48,
```
Comment (zombieJ) [inline components/upload/style/dragger.tsx:47]: `FIXME`

File: components/upload/style/index.tsx [hunk 7]
```diff
@@ -1,8 +1,59 @@
...
+    genBaseStyle(token),
+    genDraggerStyle(token),
+    genPictureStyle(token),
+    genPictureCardStyle(token),
+    genListStyle(token),
+    genMotionStyle(token, hashId),
+    genRtlStyle(token),
+  ],
+  token => ({
+    uploadPictureCardSize: 104,
```
Comment (zombieJ) [inline components/upload/style/index.tsx:55]: `FIXME`

File: components/upload/style/list.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            transition: `all ${token.motionDurationSlow}`,
+          },
+
+          [actionsCls]: {
+            [actionCls]: {
+              opacity: 0,
+            },
+
+            [`${actionCls}${antCls}-btn-sm`]: {
+              height: 20,
```
Comment (zombieJ) [inline components/upload/style/list.tsx:45]: `FIXME`

File: components/upload/style/list.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+          },
+
+          [`${componentCls}-icon ${iconCls}`]: {
+            color: token.colorTextSecondary,
+            fontSize: token.fontSizeBase,
+          },
+
+          [`${itemCls}-progress`]: {
+            position: 'absolute',
+            bottom: -12,
```
Comment (zombieJ) [inline components/upload/style/list.tsx:77]: `FIXME`

File: components/upload/style/motion.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+      `]: {
+          animationDuration: token.motionDurationSlow,
+          animationFillMode: token.motionEaseInOutCirc,
+        },
+
+        [`
+        ${inlineCls}-appear,
+        ${inlineCls}-enter
+      `]: {
+          animationName: `${uploadAnimateInlineIn.getName(hashId)}`,
```
Comment (zombieJ) [inline components/upload/style/motion.tsx:44]: 现在可以直接 uploadAnimateInlineIn 了~
下同

File: components/upload/style/picture.tsx [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+  const listCls = `${componentCls}-list`;
+  const itemCls = `${listCls}-item`;
+
+  return {
+    [`${componentCls}-wrapper`]: {
+      // ${listCls} 增加优先级
+      [`${listCls}${listCls}-picture, ${listCls}${listCls}-picture-card`]: {
+        [itemCls]: {
+          position: 'relative',
+          height: 66,
```
Comment (zombieJ) [inline components/upload/style/picture.tsx:16]: `FIXME`

File: components/upload/style/picture.tsx [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+          padding: token.paddingXS,
+          border: `${token.controlLineWidth}px ${token.uploadPictureCardBorderStyle} ${token.colorBorder}`,
+          borderRadius: token.radiusBase,
+
+          '&:hover': {
+            background: 'transparent',
+          },
+
+          [`${itemCls}-thumbnail`]: {
+            width: 48,
```
Comment (zombieJ) [inline components/upload/style/picture.tsx:26]: `FIXME`

File: components/upload/style/picture.tsx [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+            height: 48,
```
Comment (zombieJ) [inline components/upload/style/picture.tsx:27]: `FIXME`

Comment (zombieJ) [reply components/upload/style/picture.tsx:27]: 下面一些数字的地方都加一下 `FIXME`，这边不重复标注了~

Comment (afc163) [reply components/upload/style/picture.tsx:27]: 🆗

File: components/upload/Upload.tsx (hunk 4 above)
Comment (afc163) [reply components/upload/Upload.tsx:344]: 不需要了。

**Comment by zombieJ** [comment]
https://preview-34528-ant-design.surge.sh/components/upload-cn/#components-upload-demo-picture-card  照片墙的动画没了

**Comment by afc163** [comment]
图片

...

**Pull request merged**